patches/patch-xcode.js

```diff
@@ -11,26 +11,29 @@ const path = require( 'path' );
 const nodeModulesDir = path.join( __dirname, '../', 'node_modules' );
 
 const fetchRNPackageDirs = ( dir ) => {
-	const dirList = fs.readdirSync( dir );
+	const dirList = fs.readdirSync( dir, { withFileTypes: true } );
 	const packageDirs = [];
-	dirList.forEach( ( packageName ) => {
-		const packageDir = path.join( dir, packageName );
-		if ( packageName.startsWith( '@' ) ) {
-			packageDirs.push( ...fetchRNPackageDirs( packageDir ) );
-		} else {
-			const files = fs.readdirSync( packageDir );
-			const podSpecs = files.filter( ( file ) =>
-				file.toLowerCase().endsWith( '.podspec' )
-			);
-			if ( podSpecs.length > 0 ) {
-				packageDirs.push( {
-					dir: packageDir,
-					files: podSpecs,
-					package: packageName,
-				} );
+	dirList
+		.filter( ( file ) => file.isDirectory() )
+		.map( ( file ) => file.name )
+		.forEach( ( packageName ) => {
```

Comment:
Alternatively, we could omit `{ withFileTypes: true }` and check for a leading dot `.` instead.

Reply:
I think that's error prone as it would break if npm decides to change that by adding another directory without a leading dot. This patch file is already a hack and we should prevent it from breaking our build again 😅 .